gempy/core/data_modules/geometric_data.py

```diff
@@ -1033,80 +1038,116 @@ def rescale_data(self, rescaling_factor=None, centers=None):
         Returns:
 
         """
-        max_coord, min_coord = self.max_min_coord(self.surface_points, self.orientations)
+
+        xyz = self.concat_surface_points_orientations(self.surface_points.df[['X', 'Y', 'Z']],
+                                                      self.orientations.df[['X', 'Y', 'Z']])
+
+        # This is asking for XYZ parameters
+        max_coord, min_coord = self.max_min_coord(xyz)
+
         if rescaling_factor is None:
-            self.df['rescaling factor'] = self.compute_rescaling_factor(self.surface_points, self.orientations,
-                                                                        max_coord, min_coord)
+            # This is asking for XYZ parameters
+            self.df['rescaling factor'] = self.compute_rescaling_factor_for_0_1(max_coord=max_coord,
+                                                                                min_coord=min_coord)
         else:
             self.df['rescaling factor'] = rescaling_factor
         if centers is None:
-            self.df.at['values', 'centers'] = self.compute_data_center(self.surface_points, self.orientations,
-                                                                       max_coord, min_coord)
+            # This is asking for XYZ parameters
+            self.df.at['values', 'centers'] = self.compute_data_center(max_coord=max_coord,
+                                                                       min_coord=min_coord)
         else:
             self.df.at['values', 'centers'] = centers
 
-        self.set_rescaled_surface_points()
-        self.set_rescaled_orientations()
-        self.set_rescaled_grid()
+        self.set_rescaled_surface_points(axial_anisotropy=axial_anisotropy)
+        self.set_rescaled_orientations(axial_anisotropy=axial_anisotropy)
+        self.set_rescaled_grid(axial_anisotropy=axial_anisotropy)
         return True
 
+    def compute_axial_anisotropy(self, extent=None):
+        if extent is None:
+            extent = self.grid.regular_grid.extent
+
+        # Calculate average
+
+        x_ext = np.linalg.norm(extent[0]-extent[1])
+        y_ext = np.linalg.norm(extent[2]-extent[3])
+        z_ext = np.linalg.norm(extent[4]-extent[5])
+        mean_extent = np.mean([x_ext, y_ext, z_ext])
+        return np.array([mean_extent/x_ext, mean_extent/y_ext, mean_extent/z_ext])
```

Comment:
this looks a lot like my old pr for anisotropies, maybe use sklearn transforms instead here

Reply:
For adding rotation? Yes, probably we should somehow merge what you did in the old pull request but I think it is better to do it after we merge. Also, writing the interpolation on keops I saw this https://www.kernel-operations.io/keops/_auto_examples/pytorch/plot_anisotropic_kernels.html#sphx-glr-auto-examples-pytorch-plot-anisotropic-kernels-py. It is very similar idea to what we want to do here but they do the transformation on the distance calculation. That may simplify things